lib/puppet/pops/types/types.rb

```diff
@@ -90,6 +94,29 @@ def kind_of_callable?(optional=true)
         false
       end
 
+      # Returns `true` if an instance of this type is iterable, `false` otherwise
+      # The method #iterable_type must produce a `PIterableType` instance when this
+      # method returns `true`
+      #
+      # @return [Boolean] flag to indicate if instances of  this type is iterable.
+      def iterable?
+        false
+      end
+
+      # Returns the `PIterableType` that this type should be assignable to to, or `nil` if no such type exists.
+      # A type that returns a `PIterableType` must respond `true` to `#iterable?`.
+      #
+      # Any Collection[T] is assignable to an Iterable[T]
+      # A String is assignable to an Iterable[String] iterating over the strings characters
+      # An Integer is assignable to an Iterable[Integer] iterating over the 'times' enumerator
+      # A Type[T] is assignabel to an Iterable[Type[T]] if T is an Integer or Enum
+      #
```

Comment:
These are just examples, since the actual implementation that makes these true are elsewhere (in the respective types). Maybe point that out.